enhancements/proposed/20200501-json-output/start-json-output.md

```diff
@@ -57,6 +70,53 @@ If minikube fails, and an actionable error message exists, the following JSON wi
 ```
 
 ### Stdout
+If `--output json` is specified, each step of `minikube start` will be output as JSON:
+
+```json
+{
+  "Name": "Selecting Driver",
+  "Message": "✨  Using the hyperkit driver based on user configuration\n",
+  "TotalSteps": 9,
+  "CurrentStep": 2,
+  "Type": "Log"
+}
+```
+
+and each update on a downloaded artifact will be output as:
+
+```json
+{
+  "Type": "Download",
+  "Artifact": "preload.tar.gz",
+  "Progress": "10%"
+}
+```
+
+`minikube start` output would look something like this:
+
+```
+$ minikube start
+{"Name":"Minikube Version","Message":"😄  minikube v1.10.0-beta.2 on Darwin 10.14.6\n","TotalSteps":9,"CurrentStep":1, "Type":"Log"}
+{"Name":"Selecting Driver","Message":"✨  Using the hyperkit driver based on user configuration\n","TotalSteps":9,"CurrentStep":2,"Type":"Log"}
+{"Name":"Starting Control Plane","Message":"👍  Starting node minikube in cluster minikube\n","TotalSteps":9,"CurrentStep":3,"Type":"Log"}
+{"Name":"Download Necessary Artifacts","Message":"💾  Downloading Kubernetes v1.18.1 preload ...\n","TotalSteps":9,"CurrentStep":4,"Type":"Log"}
+  {"Type":"Download", "Artifact":"preload.tar.gz", "Progress": "10%"}
```

Comment:
If this message is an extension of Step 4, it would be nice to have some indication of that. It would remove the requirement that no other unrelated messages come in the middle.

Reply:
Added that in -- do you have any preference between this way and Drew's suggestion in this [comment](https://github.com/kubernetes/minikube/pull/8105#discussion_r424765969)?

Reply:
I would prefer the method shows in Drew's comment. Much easier to parse.